rbc/extension_functions.thrift

```diff
@@ -91,8 +91,9 @@ enum TOutputBufferSizeType {
 
 struct TUserDefinedFunction {
   1: string name,
-  2: list<TExtArgumentType> argTypes
-  3: TExtArgumentType retType
+  2: list<TExtArgumentType> argTypes,
+  3: TExtArgumentType retType,
+  4: bool usesManager
```

Comment:
`TUserDefinedTableFunction` uses
```
list<map<string, string>> annotations
```
than enables introducing new parameters without changing the thrift interface.
I suggest using the same approach here, for instance, via:
```suggestion
  4: list<map<string, string>> annotations
```
so that the signature `<return value>(RowFunctionManager, <arg0>, <arg1>, ...)` is expressed as `annotations=[<arg0 annotations>, <arg1 annotations>, ..., <return value annotations>, dict(uses_manager='1')]` when following the same logic as in `TableFunction::getInputAnnotations`, `TableFunction::getOutputAnnotations`, and `TableFunction::getFunctionAnnotations`.

Reply:
This requires updating the thrift in heavydb first, as I implemented with `bool usesManager` there.

Reply:
Could we change it there? We likely need `annotations` for UDFs anyway, for instance, for naming arguments etc.

Reply:
Yes, I'm doing the necessary changes there as well